llvm/lib/DebugInfo/DWARF/DWARFDebugLine.cpp

```diff
@@ -1337,7 +1339,22 @@ uint32_t DWARFDebugLine::LineTable::lookupAddressImpl(
                                       DWARFDebugLine::Sequence::orderByHighPC);
   if (It == Sequences.end() || It->SectionIndex != Address.SectionIndex)
     return UnknownRowIndex;
-  return findRowInSeq(*It, Address);
+
+  uint32_t RowIndex = findRowInSeq(*It, Address);
+
+  if (RowIndex == UnknownRowIndex)
+    return RowIndex;
+
+  // Approximation will only be attempted if a valid RowIndex exists.
+  if (Approximation && Approximation->Report) {
+    for (uint32_t ApproxRowIndex = RowIndex;
+         ApproxRowIndex >= It->FirstRowIndex; --ApproxRowIndex) {
+      if (Rows[ApproxRowIndex].Line)
+        return ApproxRowIndex;
+      Approximation->IsApproximateLine = true;
```

Comment:
You are missing the case where if the user input the address(which already has non-zero line) and also used the option ```--skip-line-zero``` then this will not be considered as an approximate output(check lit test prefix NO-APPROX).